Source/DafnyRuntime/DafnyRuntimeJava/src/main/java/dafny/DafnySequence.java

```diff
@@ -859,11 +879,13 @@ private NonLazyDafnySequence<T> computeElements() {
             if (seq instanceof ConcatDafnySequence<?>) {
                 ConcatDafnySequence<T> cseq = (ConcatDafnySequence<T>) seq;
 
-                if (cseq.ans != null) {
-                    copier.copyFrom(cseq.ans);
+                leftBuffer = cseq.left;
+                rightBuffer = cseq.right;
+                if (leftBuffer == null || rightBuffer == null) {
```

Comment:
Isn't this condition backwards? i.e.
```suggestion
                if (leftBuffer != null && rightBuffer != null) {
```

Reply:
Good catch. I flipped the if and else bodies instead to stay consistent with the other cases, though.